src/java.base/share/classes/jdk/internal/jimage/ImageReader.java

```diff
@@ -617,7 +620,7 @@ public boolean isResource() {
             return false;
         }
 
-        // TODO (review note): Sure this could/should be IllegalStateException?
+        // TODO (review note): Could/should this be IllegalStateException?
```

Comment:
IllegalStateException would be fine.  It can't/should not occur.
The exception message might more helpfully return the type of the Node, not just the node's name.
It could also be an InternalError, since a subclass of Node should have overridden getChildNames() and didn't.
But this is a tight little subclass tree so it doesn't matter.